### PR TITLE
Fixed #7182 - a bug when simulating with leap days

### LINKED ISSUE #7182: Energy Meters report is screwy when simulating with leap days
Issue overview
--------------
Run the attached IDF file using the attached EPW that includes a leap day. The Energy Meters report in the html file will look like this:

image

If you modify the EPW to turn off the leap day (e.g., change "HOLIDAYS/DAYLIGHT SAVINGS,Yes,0,0,0" to "HOLIDAYS/DAYLIGHT SAVINGS,No,0,0,0" in the EPW header), the results are fine:

image

[in.idf.txt](https://github.com/NREL/EnergyPlus/files/2886375/in.idf.txt)
[in.epw.txt](https://github.com/NREL/EnergyPlus/files/2886376/in.epw.txt)

cc @joseph-robertson @jasondegraw 

### Details
Some additional details for this issue (if relevant):
 - Platform: Windows 10
 - Version of EnergyPlus: 9.0.1

### Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [x] Defect file added (list location of defect file here)
 - [x] Ticket added to Pivotal for defect (development team task)
 - [x] Pull request created (the pull request will have additional tasks related to reviewing changes that fix this defect)

### PR DESCRIPTION
**Pull request by dareumnam**
Pull request overview
---------------------
Fixed #7182 

1. Fixed SimulationManager.cc
- A leap year has 366 days, but the checks that initialize and get things properly set up for meters checked for 365 days.

2. Add a regression test case
- Since all of the unit tests that run ManageSimulation use design days (with NumOfDayInEnvrn = 1, but I needed to run with NumOfDayInEnvrn = 366), a regression test was done instead of a unit test,
- the results of the regression test showed that the problem of issue #7182 was fixed.
- the regression test result with the original build of EnergyPlus (develop branch)
image
- the regression test result with the fixed code (this PR branch)
image

* epw file credit goes to @shorowit via #7182

 
### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue.  The github issue should be referenced in the PR description
   - Refactoring: This pull request includes code changes that don't change the functionality of the program, just perform refactoring
   - NewFeature: This pull request includes code to add a new feature to EnergyPlus
   - Performance: This pull request includes code changes that are directed at improving the runtime performance of EnergyPlus
   - DoNoPublish: This pull request includes changes that shouldn't be included in the changelog

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add to input rules file for interfaces
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Required documentation updates?
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

### CODE REVIEW

File: testfiles/CMakeLists.txt [hunk 1]
```diff
@@ -1,698 +1,699 @@
...
-ADD_SIMULATION_TEST(IDF_FILE WindACRHControl.idf EPW_FILE USA_FL_Miami.Intl.AP.722020_TMY3.epw)
-ADD_SIMULATION_TEST(IDF_FILE WindowTests.idf EPW_FILE USA_CO_Golden-NREL.724666_TMY3.epw)
-ADD_SIMULATION_TEST(IDF_FILE WindowTestsSimple.idf EPW_FILE USA_CO_Golden-NREL.724666_TMY3.epw)
-ADD_SIMULATION_TEST(IDF_FILE WWHPSimpleAuto.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw)
-ADD_SIMULATION_TEST(IDF_FILE ZoneCoupledGroundHTBasement.idf EPW_FILE USA_CO_Golden-NREL.724666_TMY3.epw)
-ADD_SIMULATION_TEST(IDF_FILE ZoneCoupledGroundHTSlabInGrade.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw)
-ADD_SIMULATION_TEST(IDF_FILE ZoneCoupledGroundHTSlabOnGrade.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw)
-ADD_SIMULATION_TEST(IDF_FILE ZoneCoupledKivaSlab.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw)
-ADD_SIMULATION_TEST(IDF_FILE ZoneCoupledKivaBasement.idf EPW_FILE USA_CO_Golden-NREL.724666_TMY3.epw)
-ADD_SIMULATION_TEST(IDF_FILE ZoneCoupledKivaWalkoutBas
```
Comment (Myoldmopar) [inline testfiles/CMakeLists.txt]: @dareumnam something went wrong with this file, it is showing every line as different.  This usually indicates the end-of-line character was messed up.  You'll want to verify the editor you were using to edit this file is set to write out the correct line endings for your platform. (Windows is CRLF line endings)

File: src/EnergyPlus/SimulationManager.cc [hunk 2]
```diff
@@ -502,10 +503,16 @@ namespace SimulationManager {
             DayOfSim = 0;
             DayOfSimChr = "0";
             NumOfWarmupDays = 0;
-            if (NumOfDayInEnvrn <= 365) {
-                isFinalYear = true;
+            if (CurrentYearIsLeapYear) {
```
Comment (Myoldmopar) [inline src/EnergyPlus/SimulationManager.cc:506]: This seems quite reasonable.

File: testfiles/CMakeLists.txt [hunk 3]
```diff
@@ -621,6 +621,7 @@ ADD_SIMULATION_TEST(IDF_FILE WCE_Diffuse_Shade.idf EPW_FILE USA_IL_Chicago-OHare
 ADD_SIMULATION_TEST(IDF_FILE WCE_DoubleClear_BSDF.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw)
 ADD_SIMULATION_TEST(IDF_FILE WCE_Interior_VB_-45_deg.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw)
 ADD_SIMULATION_TEST(IDF_FILE _5ZoneAirCooled_annual.idf EPW_FILE USA_IL_Chicago-OHare.Intl.AP.725300_TMY3.epw ANNUAL_SIMULATION)
+ADD_SIMULATION_TEST(IDF_FILE _5ZoneAirCooled_LeapYear_annual.idf EPW_FILE MadeUpLeapYear.epw ANNUAL_SIMULATION)
```
Comment (Myoldmopar) [inline testfiles/CMakeLists.txt:624]: OK, I just converted the file to the right line endings and pulled in the latest upstream develop and it's all good here again.

**Review by Myoldmopar — APPROVED** [review]

File: testfiles/CMakeLists.txt (hunk 1 above)
Comment (dareumnam) [reply testfiles/CMakeLists.txt]: I got it. I'll double-check the editor and make sure not to write a wrong end-of-line character. Thanks!

**Pull request merged**